lib/src/builder/generation/declaration_parsing.dart

```diff
@@ -12,35 +12,66 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-library over_react.transformer.declaration_parsing;
-
-import 'dart:mirrors';
 
 import 'package:analyzer/analyzer.dart';
-import 'package:barback/barback.dart' show TransformLogger;
+import 'package:logging/logging.dart';
+import 'package:over_react/src/builder/util.dart';
 import 'package:over_react/src/component_declaration/annotations.dart' as annotations;
-import 'package:over_react/src/transformer/util.dart' show getMetaField;
 import 'package:source_span/source_span.dart';
-import 'package:transformer_utils/src/transformed_source_file.dart' show getSpan;
-import 'package:transformer_utils/transformer_utils.dart';
+import 'package:transformer_utils/transformer_utils.dart' show getSpan, NodeWithMeta;
+
+enum PropsClassType {
+  standard,
+  abstract,
+  mixin,
+  none,
+}
 
-/// A set of [NodeWithMeta] component pieces declared using `over_react` transformer annotations.
+/// A set of [NodeWithMeta] component pieces declared using `over_react` builder annotations.
 ///
 /// Can include:
 ///
 /// * A single component declared using a `@Factory()`, `@Component()`, `@Props()`, and optionally a `@State()`
 /// * Any number of abstract component pieces: `@AbstractComponent()`, `@AbstractProps()`, `@AbstractState()`
 /// * Any number of mixins: `@PropsMixin()`, `@StateMixin()`
 class ParsedDeclarations {
-  factory ParsedDeclarations(CompilationUnit unit, SourceFile sourceFile, TransformLogger logger) {
-    const companionPrefix = r'_$';
+  factory ParsedDeclarations(CompilationUnit unit, SourceFile sourceFile, Logger logger) {
     bool hasErrors = false;
+    bool hasDeclarations = false;
 
     void error(String message, [SourceSpan span]) {
       hasErrors = true;
-      logger.error(message, span: span);
+      logger.severe(messageWithSpan(message, span: span));
     }
 
+    void validateMetaField(ClassDeclaration cd, String expectedType) {
+        final metaField = getMetaField(cd);
+        if (metaField == null) return;
+
+        if (metaField.fields.type?.toSource() != expectedType) {
+          error(
+              'Static meta field in accessor class must be of type `$expectedType`',
+              getSpan(sourceFile, metaField),
+          );
+        }
+        final isClassPrivate = cd.name.name.startsWith('_');
```

Comment:
It should be called on the companion class and mixins only. That being said, the logic for `isClassPrivate` needs to be updated to accomodate private mixins, since they'll be something like `$_SomePrivateMixin`. 

This method is only handy for the transition boilerplate. For Dart 2 only boilerplate, it's unused. Tests will be created to verify that the builder generates the field/companion class properly.

I'll add a doc comment and fix for mixins.